app/src/org/commcare/dalvik/application/CommCareApp.java

```diff
@@ -246,6 +276,10 @@ public SQLiteDatabase getHandle() {
 
     public void writeInstalled() {
         record.setStatus(ApplicationRecord.STATUS_INSTALLED);
+        record.setUniqueId(getUniqueId());

```

Comment:
you explicitly set this currently when doing the upgrade, is there ever a circumstance when this would need to get called during the install process? Is this the default way things are set?

It seems dangerous to me that they are all configured against the profile, but by fetching it and transfering over each property on its own.

Maybe it would make sense to just call something like

"record.initProfileProperties(getprofile())"

and have it do each one so it is clear that there is a one-time transfer?

The pattern here is something like:

foo.setBar(getBar())

getBar() {
    if(foo.getBar() == something){
       return soemthing
    }
    return foo.Bar();
}

so the model both depends on _and_ sets the value of a property, which in my mind should only happen once.